pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginManager.java

```diff
@@ -229,22 +257,86 @@ private void initRecordReaderClassMap() {
   /**
    * Loads jars recursively
    * @param pluginName
-   * @param directory
+   * @param directory the directory of one plugin
    */
   public void load(String pluginName, File directory) {
-    LOGGER.info("Trying to load plugin [{}] from location [{}]", pluginName, directory);
-    Collection<File> jarFiles = FileUtils.listFiles(directory, new String[]{"jar"}, true);
-    Collection<URL> urlList = new ArrayList<>();
-    for (File jarFile : jarFiles) {
+    Path pluginPropertiesPath = directory.toPath().resolve(PINOUT_PLUGIN_PROPERTIES_FILE_NAME);
+    if (Files.isRegularFile(pluginPropertiesPath)) {
+      Properties pluginProperties = new Properties();
+      PinotPluginConfiguration config = null;
+      try (Reader reader = Files.newBufferedReader(pluginPropertiesPath)) {
+        pluginProperties.load(reader);
+        config = new PinotPluginConfiguration(pluginProperties);
+      } catch (IOException e) {
+        LOGGER.warn("Failed to load plugin properties from {}", pluginPropertiesPath, e);
```

Comment:
Shouldn't we just fail in this case? I can imagine a rare case where Pinot doesn't have permissions to read the file. Would we want to load the plugin without reading the properties? I think I would prefer to fail fast at startup time instead.

Reply:
BTW, by failing here we can ensure that `config` is not null.

Reply:
Yep, unchecked is fine